site/book/05-developing-functions/02-developing-in-Go.md

```diff
@@ -1,133 +1,177 @@
+    return true, nil
+}
 ```
 
-Fetch the wordpress package:
+Learn more about the `KubeObject` from the [go doc](https://pkg.go.dev/github.com/GoogleContainerTools/kpt-functions-sdk/go/fn)
 
-```shell
-$ kpt pkg get https://github.com/GoogleContainerTools/kpt.git/package-examples/wordpress@v0.8
-```
 
-Test it by running the function imperatively:
+### Test the KRM function
+
+The "get-started" package contains a `./data` directory. You can use this to test out your functions. 
 
 ```shell
-$ kpt fn eval wordpress --exec ./my-fn -- myannotation=foo
-```
+# Edit the `data/resources.yaml` with your KRM resources. 
+# resources.yaml already has a `Deployment` and `Service` as test data. 
+vim data/resources.yaml
 
-During iterative development, `--exec` flag can be used to execute the
-function binary directly instead of requiring the function to be containerized
-first. Once you have a function binary that works, you can then proceed to
-creating the container image.
+# Convert the KRM resources and FunctionConfig resource to `ResourceList`.
+kpt fn source data
+
+# Pass the ResourceList as StdIn to your function
+kpt fn source data | go run main.go
 
-### Publish the function
+# Verify the KRM function behavior in the StdOutput `ResourceList`
+```
 
-Generate a Dockerfile for the function image:
+### Build the KRM function as an executable
 
+After editing your function, you can release it as an executable
 ```shell
-$ go run ./main.go gen ./
+go build -o ${FUNCTION_NAME}
+# Change permission mode if needed
+chmod +x ${FUNCTION_NAME}
 ```
 
-Build the image:
-
+#### Run the executable imperatively via kpt
 ```shell
-$ docker build . -t gcr.io/project/fn-name:tag
+kpt fn eval ./data --exec ./${FUNCTION_NAME} 
 ```
 
-Optionally, push the image to a container registry:
+#### Run the executable declaratively via kpt
+
+You can add the function to `Kptfile` via `--save` flag
 
 ```shell
-$ docker push gcr.io/project/fn-name:tag
+# Initialize ./data/Kptfile 
+kpt pkg init data
+# Add your function to the Kptfile pipeline
+kpt fn eval ./data --type mutator --save --exec ./${FUNCTION_NAME}
+
+# Run the executable in the Kptfile pipeline
+kpt fn render --allow-exec ./data
 ```
 
-Run the function imperatively as a container function:
+Once you have a function binary that works, you can then proceed to
+creating the container image.
+
+### Build the KRM function as a Docker image
+
+Build the image
+
+The "get-started" package provides the `Dockerfile`.
 
 ```shell
-$ kpt fn eval wordpress -i gcr.io/project/fn-name:tag -- myannotation=foo
+export FN_CONTAINER_REGISTRY=<Your GCR or docker hub>
+export TAG=<Your KRM function tag>
+docker build . -t ${FN_CONTAINER_REGISTRY}/${FUNCTION_NAME}:${TAG}
 ```
 
-## Next Steps
+To verify the image using the same `./data` resources
+```shell
+kpt fn eval ./data --image ${FN_CONTAINER_REGISTRY}/${FUNCTION_NAME}:${TAG}
+```
```

Comment:
Generally it's a good idea to have a closing comment with links to resources where they can learn more about. Here you can leave links for `set-namespace` function codebase, `fn-sdk` etc.

Reply:
sounds good. I added a link to point to the kpt-functions-catalog contributor guide. I think we can also add some more complicated examples in this section later on.